README.md

```diff
@@ -0,0 +1,103 @@
+# godockerize
+
+godockerize builds a Dockerfile for a given Go project. The base image is Alpine
+3.8.
+
+### Usage
+
+```
+$ godockerize build github.com/path/to/your/go/main
+```
+
+Will generate a Dockerfile, build a Go binary for that package, and then build
+the Dockerfile with the Go binary.
+
+Run `--help` to view the full list of options for customizing the program's
+behavior.
+
+### Build tags
+
+godockerize supports a number of build tags that can be used to customize the
+content of the generated Dockerfile. Add a `//docker:<tagname>` comment line to
+the package being built to customize the behavior. Here are the tags:
+
+##### //docker:env
+
+Adding this to your Go binary:
+
+```go
+//docker:env TZ=UTC
+
+package main
+```
+
+Adds an ENV command to the dockerfile:
+
+```
+ENV TZ=UTC
+```
+
+##### //docker:expose
+
+Expose the provided ports via an EXPOSE command
+
+```go
+//docker:expose 5000
+```
+
+##### //docker:install
+
+Install the provided packages. In addition, if a package name ends with `@edge`,
+the edge apk repositories will be added to `/etc/apk/repositories`.
+
+```go
+//docker:install git@edge openssh-client
+```
+
+Generates:
+
+```
+  FROM alpine:3.8
+  RUN echo -e "@edge http://dl-cdn.alpinelinux.org/alpine/edge/main" >> /etc/apk/repositories
+  RUN echo -e "@edge http://dl-cdn.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories
+  RUN apk add --no-cache ca-certificates git@edge mailcap openssh-client tini
+  USER myuser
+  ENTRYPOINT ["/sbin/tini", "--", "/usr/local/bin/mybinary"]
+  ADD mybinary /usr/local/bin/
+```
+
+##### //docker:repository
+
+Can be used to add other repository versions.
```

Comment:
It is probably worth documenting that the `edge` repo is automatically added as a special case for any dependencies tagged as `@edge`.